go.mod

```diff
@@ -10,15 +10,15 @@ require (
 	github.com/onsi/gomega v1.13.0
 	github.com/pkg/errors v0.9.1
 	github.com/spf13/pflag v1.0.5
-	golang.org/x/oauth2 v0.0.0-20201203001011-0b49973bad19
-	k8s.io/api v0.21.1
-	k8s.io/apimachinery v0.21.1
-	k8s.io/client-go v0.21.1
+	golang.org/x/oauth2 v0.0.0-20210615190721-d04028783cf1
+	k8s.io/api v0.21.2
+	k8s.io/apimachinery v0.21.2
+	k8s.io/client-go v0.21.2
 	k8s.io/klog/v2 v2.9.0
 	k8s.io/utils v0.0.0-20210527160623-6fdb442a123b
-	sigs.k8s.io/cluster-api v0.4.0-beta.1
-	sigs.k8s.io/cluster-api/test v0.4.0-beta.1
-	sigs.k8s.io/controller-runtime v0.9.0
+	sigs.k8s.io/cluster-api v0.4.0
+	sigs.k8s.io/cluster-api/test v0.4.0
+	sigs.k8s.io/controller-runtime v0.9.1
 )
 
-replace sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v0.4.0-beta.1
+replace sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v0.4.0
```

Comment:
I think this replacement is not needed anymore

Reply:
go this:
```
$ go mod tidy
go: sigs.k8s.io/cluster-api/test@v0.4.0 requires
        sigs.k8s.io/cluster-api@v0.0.0-00010101000000-000000000000: invalid version: unknown revision 000000000000
```

lets keep as-is for now and I will follow up in CAPI

Reply:
I see, seems because this one 
https://github.com/kubernetes-sigs/cluster-api/blob/7f879be68d15737e335b6cb39d380d1d163e06e6/test/go.mod#L26